File: 242-valid-anagram/242-valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t): return False
        
        mp = collections.defaultdict(int)
        
        for i in range(len(s)):
            mp[ord(s[i]) - ord('a')] += 1
            mp[ord(t[i]) - ord('a')] -= 1
            
        for _, count in mp.items():
            if count != 0: return False
        return True

File: 242-valid-anagram/242-valid-anagram.cpp
#include <unordered_map>

using namespace std;

class Solution {
public:
    bool isAnagram(string s, string t) {
        if (s.length() != t.length())
            return false;
        unordered_map<char, int> map;
        
        for (int i = 0; i < s.length(); i++) {
            if (!map.count(s[i]))
                map[s[i]] = 0;
            if (!map.count(s[i]))
                map[t[i]] = 0;
            map[s[i]] += 1;
            map[t[i]] -= 1;
        }
        
        for (auto item : map)
            if (item.second != 0)
                return false;
        
        return true;
    }
};